[bug]: invalid memory address or nil pointer dereference

### Background

After upgrading my lnd node to `v0.17.0-beta.rc1`, my node fails to startup with a segmentation violation.

### Your environment

* version of `lnd`: `v0.17.0-beta.rc1`
* which operating system (`uname -a` on *Nix): `Linux 5.10.0-23-amd64 #1 SMP Debian 5.10.179-1 (2023-05-12) x86_64 GNU/Linux`
* version of `btcd`, `bitcoind`, or other backend: `bitcoind`
* any other relevant environment details

### Steps to reproduce

```sh
lnd --accept-keysend --profile 6061 --tor.active --tor.v3 --protocol.wumbo-channels --db.prune-revocation
...
lncli unlock
```

### Expected behaviour

I expect lnd to startup without a segmentation violation.

### Actual behaviour

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xcf7a9e]

goroutine 1 [running]:
github.com/lightningnetwork/lnd/contractcourt.(*ChannelArbitrator).relaunchResolvers(0xc000598580, 0x0, 0x0?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/channel_arbitrator.go:739 +0x6fe
github.com/lightningnetwork/lnd/contractcourt.(*ChannelArbitrator).Start(0x146e4a0?, 0xc001406e40)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/channel_arbitrator.go:560 +0x665
github.com/lightningnetwork/lnd/contractcourt.(*ChainArbitrator).Start(0xc002771e80?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/chain_arbitrator.go:707 +0x11fe
github.com/lightningnetwork/lnd.(*server).Start.func1()
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/server.go:1987 +0x19ff
sync.(*Once).doSlow(0x3?, 0x16d9432?)
        /usr/lib/go-1.19/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/lib/go-1.19/src/sync/once.go:65
github.com/lightningnetwork/lnd.(*server).Start(0xc4998?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/server.go:1862 +0x76
github.com/lightningnetwork/lnd.Main(0xc0003fa580, {{0x0?, 0xc000006340?, 0xc000066738?}}, 0xc000120fc0, {0xc0000906c0, 0xc000100720, 0xc000100780, 0xc0001007e0, {0x0}})
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/lnd.go:676 +0x3df4
main.main()
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/cmd/lnd/main.go:38 +0x205
```

@GeorgeTsagk thank you for your hint, but with #7936 lnd still crashes on startup with a slightly different segmentation violation:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xcf7ab9]

goroutine 1 [running]:
github.com/lightningnetwork/lnd/contractcourt.(*ChannelArbitrator).relaunchResolvers(0xc00062b080, 0x0, 0x9660fd0?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/channel_arbitrator.go:741 +0x6f9
github.com/lightningnetwork/lnd/contractcourt.(*ChannelArbitrator).Start(0x146e4a0?, 0xc0030556f0)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/channel_arbitrator.go:563 +0x665
github.com/lightningnetwork/lnd/contractcourt.(*ChainArbitrator).Start(0xc00587a100?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/contractcourt/chain_arbitrator.go:714 +0x11fe
github.com/lightningnetwork/lnd.(*server).Start.func1()
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/server.go:1987 +0x19ff
sync.(*Once).doSlow(0x3?, 0x16d9432?)
        /usr/lib/go-1.19/src/sync/once.go:74 +0xc2
sync.(*Once).Do(...)
        /usr/lib/go-1.19/src/sync/once.go:65
github.com/lightningnetwork/lnd.(*server).Start(0xc49b3?)
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/server.go:1862 +0x76
github.com/lightningnetwork/lnd.Main(0xc000586000, {{0x0?, 0xc000006340?, 0xc000066738?}}, 0xc000122e40, {0xc0000a09c0, 0xc00007e540, 0xc00007e5a0, 0xc00007e600, {0x0}})
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/lnd.go:676 +0x3df4
main.main()
        /home/lnd-mainnet/gocode/src/github.com/lightningnetwork/lnd/cmd/lnd/main.go:38 +0x205
```

--- a/contractcourt/channel_arbitrator.go
+++ b/contractcourt/channel_arbitrator.go
@@ -731,13 +731,14 @@ func (c *ChannelArbitrator) relaunchResolvers(commitSet *CommitSet,
 
 		if chanState != nil {
 			resolver.SupplementState(chanState)
-		}
 
-		// For taproot channels, we'll need to also make sure the
-		// control block information was set properly.
-		maybeAugmentTaprootResolvers(
-			chanState.ChanType, resolver, contractResolutions,
-		)
+			// For taproot channels, we'll need to also make sure
+			// the control block information was set properly.
+			maybeAugmentTaprootResolvers(
+				chanState.ChanType, resolver,
+				contractResolutions,
+			)
+		}
 
 		unresolvedContracts[i] = resolver
 

